README.md

```diff
@@ -64,17 +64,30 @@ class MyDataSource(panoply.DataSource):
 
 The `DataSource` base class exposes the following methods:
 
-###### __init__(self, source, options)
+###### __init__(self, source, options, events)
 
-Constructor. Receives a dictionary with the data `source` parameters (see below) and a dictionary with any additional `options`. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc.
+Constructor. Receives a dictionary with the data `source` parameters (see below), a dictionary with any additional `options` and a dictionary with `events` subscriptions. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc, and it may contain additional parameters that can't be transferred with the source
+The `events` parameter signs event handlers for this data source. It is a dictionary with its keys being the event type (or '*' for all), and its values being a list of functions to call when the event arrives.
 
-**Make sure** to call `super()` if you need to override it.
+`source` and `options` are available as attributes from within the class instance.
+
+`source` should have a `destination` key (String) describing the default table destination for data retrieved using the data source. The default destination is the type of the data source as will be defined by Panoply
```

Comment:
👍 
- add . at the end
- explain what is destination. again - remember this sdk is _not only_ for us. as internal user, it will be nice to understand that destination is name of the redshift table where my results swill be saved.